src/main/java/com/capitalone/dashboard/collector/DefaultWhiteSourceClient.java

```diff
@@ -0,0 +1,316 @@
+import org.apache.commons.logging.LogFactory;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+@Component
+public class DefaultWhiteSourceClient implements WhiteSourceClient {
+    private static final Log LOG = LogFactory.getLog(DefaultWhiteSourceClient.class);
+    private static final String API_URL = "/api/v1.3";
+    private static final String DATE_FORMAT = "yyyy-MM-dd HH:mm:ss";
+    private final RestClient restClient;
+    private final WhiteSourceSettings whiteSourceSettings;
+
+    @Autowired
+    public DefaultWhiteSourceClient(RestClient restClient, WhiteSourceSettings settings) {
+        this.restClient = restClient;
+        this.whiteSourceSettings = settings;
+    }
+
+    @Override
+    public String getOrgDetails(String instanceUrl, String orgToken) {
+        try {
+            JSONObject jsonObject = makeRestCall(getApiBaseUrl(instanceUrl), Constants.RequestType.getOrganizationDetails, orgToken, null, null);
+            String orgName = (String) jsonObject.get(Constants.ORG_NAME);
+            return orgName;
+        } catch (Exception e) {
+            LOG.error("Exception occurred while calling getOrgDetails " + e.getStackTrace());
+        }
+        return null;
+    }
+
+    @Override
+    public List<WhiteSourceProduct> getProducts(String instanceUrl, String orgToken) {
+        List<WhiteSourceProduct> whiteSourceProducts = new ArrayList<>();
+        try {
+            JSONObject jsonObject = makeRestCall(getApiBaseUrl(instanceUrl), Constants.RequestType.getAllProducts, orgToken, null, null);
+            if (Objects.isNull(jsonObject)) return new ArrayList<>();
+            JSONArray jsonArray = (JSONArray) jsonObject.get(Constants.PRODUCTS);
+            if (CollectionUtils.isEmpty(jsonArray)) return new ArrayList<>();
+            for (Object product : jsonArray) {
+                JSONObject wsProduct = (JSONObject) product;
+                WhiteSourceProduct whiteSourceProduct = new WhiteSourceProduct();
+                whiteSourceProduct.setProductId(getLongValue(wsProduct, Constants.PRODUCT_ID));
+                whiteSourceProduct.setProductName(getStringValue(wsProduct, Constants.PRODUCT_NAME));
+                whiteSourceProduct.setProductToken(getStringValue(wsProduct, Constants.PRODUCT_TOKEN));
+                whiteSourceProducts.add(whiteSourceProduct);
+            }
+        } catch (Exception e) {
+            LOG.error("Exception occurred while retrieving getAllProducts " + e.getMessage());
```

Comment:
Again, don't silently absorb the exception, as the logic at caller will not be correct or NPE. Throw the exception out, catch it in the caller's loop, and skip to the next org.

Reply:
this logic iterates through all products in org and should not skip to next org until all the products information is collected. If in the process of collecting product information, exceptions should be logged and should continue to next product. I can add more logging in catch block with product name.

Reply:
yes, this call can skip to next org and exit the process. Adding throws Exception.